### ISSUE TITLE
MATCH query bug on values containing whitespace

### ISSUE THREAD

**Issue by creisle** [orientechnologies/orientdb#8400]
## Expected behavior  
I can select the record by querying name = 'the name' but then when I try to query using a match query the query fails. I am selecting a record that has both in and out going edges.

The query that works looks something like
```
select * from myclass where name = 'the name'
```
the match query that fails looks something like
```
MATCH {class: myclass, where: (name = 'the name')}.out(){while: (out().size() > 0)} RETURN $pathElements
```
I expected the match to return itself and related records

## Actual behavior  
the select works but the match returns no records

## Steps to reproduce  
This is difficult to reproduce. I only seem to be able to do it when there is a significant number of entries in a field with a FULL_TEXT_HASH index. It also only affects records where that field name contains a space (or likely any other separator char). The smallest set I have so far been able to reproduce it with is 800+ vertices and 800+ edges

If I delete the full text hash the problem goes away. I think it is an issue with the match query using the full text hash for = (I thought it would only use it for containstext queries)

**Comment by creisle** [comment]
ok so I think I figured out how to easily recreate this.

```
create property V.name string
create index hashnames on V (name) fulltext_hash_index
create vertex v set name = 'nospaces'
create vertex v set name = 'has some spaces'
```

Currently the match query will work (will return the single 'has some spaces' record)
```
MATCH {class: v, where: (name = 'has some spaces')}.out(){while: (out().size() > 0)} RETURN $pathElements
```

but then if we continue to add records (repeat the SQL statements based on the bracketed xN numbers)
```
create vertex v set name = 'has some spaces'  (x 20)
create vertex v set name = 'somethinga little more complicated' (x21)
```

and then try the match query again.... it will now fail

**Comment by luigidellaquila** [comment]
Hi @creisle 

Thank you very much, this helps a lot.
I managed to reproduce the issue, I'll try to fix it asap

Thanks

Luigi

**Comment by luigidellaquila** [comment]
Hi @creisle 

I just pushed a fix for this, it will be released with v 2.2.37

Thanks

Luigi
